WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -172,6 +173,11 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 					HdPubKey foundKey = KeyManager.GetKeyForScriptPubKey(output.ScriptPubKey);
 					if (foundKey is { })
 					{
+						if (!foundKey.IsInternal)
+						{
+							tx.Label = SmartLabel.Merge(tx.Label, foundKey.Label);
+						}
```

Comment:
This makes sure that the tx gets the label of the receive address only and not the change.